packages/app-project/stores/User/UserPersonalization/UserProjectPreferences/UserProjectPreferences.js

```diff
@@ -97,13 +89,29 @@ const UserProjectPreferences = types
       },
 
       fetchResource: flow(function* fetchResource() {
-        if (!self.id) {
-          yield _fetch()
+        try {
+          if (!self.id) {
+            self.setLoadingState(asyncStates.loading)
+            const preferences = yield _fetch()
+            if (preferences) {
+              self.setResource(preferences)
+            }
+            self.setLoadingState(asyncStates.success)
+          }
+        } catch (error) {
+          self.handleError(error)
         }
       }),
 
       refreshResource: flow(function * refreshResource() {
-        yield _fetch()
+        try {
+          const preferences = yield _fetch()
+          if (preferences) {
+            self.preferences = preferences.preferences
+          }
+        } catch (error) {
+          self.handleError(error)
+        }
```

Comment:
This will crash the classifier if the refresh request fails (eg. I'm on mobile and temporarily lose network.) It might be better to ignore the error and let the volunteer continue to classify with their old UPP.

Reply:
I think it might be better to ignore the error and let the volunteer continue to classify.

Reply:
I've changed this to log the error without changing the loading state.